### DISCUSSION TITLE
Why can object initializes assign 'init' but not 'readonly'

### DISCUSSION THREAD

**Discussion by zms9110750** [dotnet/csharplang#8409] [General] [answered]
Isn't init used as a substitute for readonly on property? Why does he have an additional permission?

**Comment by huoyaoyuan** [comment, marked as answer]
> Isn't init used as a substitute for readonly on property?

No. Getter-only auto properties corresponds to `readonly` fields. They can be assigned in constructor.

> Why does he have an additional permission?

`readonly` means it can only be set by the object itself, in constructor. `init` allows initialization from the caller that uses the object.

**Reply by jnm2** [reply]
> `readonly` means it can only be set by the object itself, in constructor.

Or, in any `init` accessor.
